pkgs/games/wargus/default.nix

```diff
@@ -1,27 +1,26 @@
 { stdenv, lib, callPackage, fetchFromGitHub
-, fetchurl, runCommand, unzip, bchunk, p7zip
+, requireFile, runCommand, p7zip
 , cmake, pkg-config, makeWrapper
 , zlib, bzip2, libpng
-, dialog, python3, cdparanoia, ffmpeg
+, ffmpeg
 }:
 
 let
   stratagus = callPackage ./stratagus.nix {};
 
-  dataDownload = fetchurl {
-    url = "https://archive.org/download/warcraft-ii-tides-of-darkness_202105/Warcess.zip";
-    sha256 = "0yxgvf8xpv1w2bjmny4a38pa3xcdgqckk9abj21ilkc5zqzqmm9b";
+  dataDownload = requireFile {
+    message = ''
+      Provide a Warcraft II CD-ROM image.
+    '';
+    name = "WC2BTDP01.iso";
+    sha256 = "0wc4wqb9afxykah8lq1jw0sl564j5gs1shrr67cflfsaiscr4qxm";
```

Comment:
This incorrectly assumes the archive.org iso is the only possible hash one might obtain for a given CD-ROM extract. It also looks like gog distributes an exe and google returns more than one possible iso that will work. (usually as some sort of rip that needs to be `bchunk`ed into an iso)

Reply:
We could make the `requireFile` overridable. Please advise a solution.

Reply:
I've been digging around for one and I'm really not sure at this point. Here's what I've pulled out so far:

Wargus assumes `${out}/share/games/stratagus/wargus` will have the game files needed. If currently gets that path via a fall back to `CMAKE_INSTALL_PREFIX/share/games/stratagus/wargus`. ([see usage](https://github.com/Wargus/wargus/blob/10084d85db36c3669da493a2bf5ba2143a4d4791/wargus.c#L68-L70) and [cmake definition](https://github.com/Wargus/wargus/blob/560192cced1b45998ebf870cc33d1dd348ec77e0/CMakeLists.txt#L154) + [fallback](https://github.com/Wargus/wargus/blob/560192cced1b45998ebf870cc33d1dd348ec77e0/CMakeLists.txt#L216)) 

The default install without game files writes files to `SHAREDIR` (`${out}/share/games/stratagus/wargus`). 

`wartool` is happy used to supply various versions the warcraft 2 files from expansion or non-expansion 'capable' export folders. It's also happy to be told it's output path.

A threads that might lead to a solution:
- configure wargus to compile in a reference to somewhere in `/var/` by default and user configurable by override.

Reply:
I can't see installing the CD files to the store and staying unfree as possible with a single definition. I can only really see adding a module that references users files as way to manage automating the call to `wartool`.

Reply:
Another thread would be to expect the CD data as an input defaulted to null. Leave it up to users and/or documentation to provide a package with the cd data as it's contents. Fail similar to requireFile without the need for an upfront hash.

Reply:
How did you get the game + source working in your original PR? https://github.com/NixOS/nixpkgs/pull/98665#issuecomment-783304642 makes it sound like I've missed something about how to get the engine configured with game files. (no issues with wargus trying to write to the store there)

Reply:
Is the original PR not working for you? It still does for me. It took a few days to figure it out.

Reply:
I'm not sure how a manual installation is supposed to go step by step. I can edit the nixpkgs install process to pull in other disk images (or the expansion image) but can't quite piece together how a normal manual installation would go (what commands when). My previous question is interested in the steps after installing the wargus package for that older version since it was manual (no wartool invocation during install).

If i delete the call to wartool in the current PR what commands would I need to run to get a working wargus install?

Reply:
You'll need to instruct the user to run wartool on the datafiles themselves.

Reply:
Here's what I've got at the moment: https://github.com/anund/nixpkgs/commit/4336d0fd3fbb74a1cc9602d4883fe185d59e5b37

If I build and run that version of staratgus + wargus without the overrides I get a version that requires a user to extract game data themselves. When I attempt to do so wargus generates a few different errors. As I'm not familiar with the package I can't distinguish the package is now broken from making mistakes manually invoking wartool/wargus.

Rephrasing my question: What is the list of commands I should be able to run to get a working copy of wargus without the disk installed as part of the nixpkgs definition?